### PR TITLE
minor improvements to plugin java api

### PR DESCRIPTION
**Pull request by ketan**

### CODE REVIEW

File: config/config-api/src/com/thoughtworks/go/domain/config/ConfigurationProperty.java [hunk 1]
```diff
@@ -187,6 +187,10 @@ public ConfigErrors errors() {
         return configErrors;
     }
 
+    public boolean hasErrors(){
+        return !configErrors.isEmpty();
```
Comment (srinivasupadhya) [inline config/config-api/src/com/thoughtworks/go/domain/config/ConfigurationProperty.java]: is this being used somewhere?

File: server/src/com/thoughtworks/go/server/service/tasks/PluggableTaskService.java [hunk 2]
```diff
@@ -75,5 +75,7 @@ public void validate(final PluggableTask modifiedTask) {
         for (ValidationError validationError : validationResult.getErrors()) {
             modifiedTask.getConfiguration().getProperty(validationError.getKey()).addError(validationError.getKey(), validationError.getMessage());
         }
+
+        return validationResult.isSuccessful();
```
Comment (srinivasupadhya) [inline server/src/com/thoughtworks/go/server/service/tasks/PluggableTaskService.java]: is this being used somewhere?

File: config/config-api/src/com/thoughtworks/go/domain/config/ConfigurationProperty.java (hunk 1 above)
Comment (ketan) [inline config/config-api/src/com/thoughtworks/go/domain/config/ConfigurationProperty.java]: The next set of commits use it, it is used in one of the (new) controllers to validate config properties of plugin tasks in an AJAXy manner.

File: server/src/com/thoughtworks/go/server/service/tasks/PluggableTaskService.java (hunk 2 above)
Comment (ketan) [inline server/src/com/thoughtworks/go/server/service/tasks/PluggableTaskService.java]: Same as above, used by the next set of commits.

**Comment by juhijariwala** [comment]
Verified on Go Version: 15.3.0(2682-566a58ea0395320403d43b5adb10ca6f99c308e6).
Works fine.

**Pull request merged**